x-pack/plugins/security_solution/public/detection_engine/rule_management_ui/pages/coverage_overview/helpers.ts

```diff
@@ -43,3 +41,20 @@ export const populateSelected = (
   allOptions.map((option) =>
     selected.includes(option.label) ? { ...option, checked: 'on' } : option
   );
+
+export const getTotalRuleCount = (
+  technique: CoverageOverviewMitreTechnique,
+  activity?: CoverageOverviewRuleActivity[]
+): number => {
+  if (!activity) {
+    return technique.enabledRules.length + technique.disabledRules.length;
+  }
+  let totalRuleCount = 0;
+  if (activity.includes(CoverageOverviewRuleActivity.Enabled)) {
+    totalRuleCount += technique.enabledRules.length;
+  }
+  if (activity.includes(CoverageOverviewRuleActivity.Disabled)) {
+    totalRuleCount += technique.disabledRules.length;
+  }
+  return totalRuleCount;
+};
```

Comment:
This is actually our domain model, it's not just a helper. It's a primitive calculation on `CoverageOverviewMitreTechnique`, or you could say its "behavior". Since it's highly coupled with `CoverageOverviewMitreTechnique`, it should be placed closer to this type under `x-pack/plugins/security_solution/public/detection_engine/rule_management/model/coverage_overview`, or even to the same file.